torch/_inductor/ir.py

```diff
@@ -6464,7 +6464,16 @@ def go_outer():  # type: ignore[no-untyped-def]
                     # individual output arguments are bound by
                     # generate_c_shim_fallback_kernel
                     if len(self.outputs) == 1:
-                        return go(self.outputs[0].get_name(), keypath)
+                        out = self.outputs[0]
+                        # When fallback kernel returns a list consisting of a single tensor,
+                        # the output is represented as a MultiOutput with non empty indices.
+                        # In this case, we strip the first key path away.
+                        return go(
+                            self.outputs[0].get_name(),
+                            keypath[1:]
+                            if isinstance(out, MultiOutput) and len(out.indices) != 0
```

Comment:
Is there a better way of doing this?

Reply:
could you instead change the `if` condition to be `if len(self.outputs) == 1 and not isinstance(self.outputs[0], MultiOutput) ...`

Reply:
The other alternative is that instead of creating a MultiOutput, we create a TensorBox (how to do that?) when the output is only a single tensor.